### PR TITLE
Movie Rating Statistics Aggregation

### PR DESCRIPTION
**Pull request by clarencecastillo**

### CODE REVIEW

File: controllers/review.controller.js [hunk 1]
```diff
@@ -29,27 +30,54 @@ class ReviewController {
...
 
-    create(author, movieId, review) {
-        return this.reviewRepository.create({
+    async updateMovieRating(movieId) {
+        const reviews = await this.getAllByMovieId(movieId);
+        const weight = 1 / reviews.length;
+        const stats = reviews.reduce((statistics, r) => {
+            statistics.count += 1;
+            statistics.average += r.rating * weight;
+            statistics.distribution[r.rating - 1] += 1;
```
Comment (jgoh120) [inline controllers/review.controller.js:39]: The distribution function shifts the rating into their respective slot?

File: controllers/review.controller.js [hunk 2]
```diff
@@ -29,27 +30,54 @@ class ReviewController {
...
+            upsert: true
+        });
+    }
+
+    async create(author, movieId, review) {
+        await this.reviewRepository.create({
             movieId: movieId,
             author: {
                 id: author.id,
                 name: `${author.firstname} ${author.lastname}`
```
Comment (jgoh120) [inline controllers/review.controller.js:61]: Why is there "{}'

File: controllers/review.controller.js (hunk 1 above)
Comment (clarencecastillo) [reply controllers/review.controller.js:39]: This line increments the appropriate counter for that number of stars eg. for a 5 star rating, we will increase the counter at position 4 by 1.

File: controllers/review.controller.js (hunk 2 above)
Comment (clarencecastillo) [reply controllers/review.controller.js:61]: This is string interpolation, where we can insert variables into a string.

**Review by jgoh120 — APPROVED** [review]

**Pull request merged**